packages/react/react/src/hooks/useLazyImage.ts

```diff
@@ -0,0 +1,67 @@
+import { useState } from 'react';
+import { useRefEffect } from './useRefEffect';
+
+interface Props {
+  src: string;
+  threshold?: number | number[];
+  root?: Document | Element | null;
+  rootMargin?: string;
+  onInView?: () => void;
+}
+
+/** @tossdocs-ignore */
+export function useLazyImage({ src, rootMargin, threshold, root, onInView }: Props) {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  const ref = useRefEffect<HTMLImageElement>(element => {
+    if (typeof IntersectionObserver === 'undefined') {
+      element.src = src;
+      return;
+    }
+
+    if (element.getAttribute('src')) {
+      console.warn('If the "src" attribute is initially in an "img" tag, lazy load is not applied.');
+      return;
+    }
+
+    const insertImageSrc = ([entry]: IntersectionObserverEntry[], observer: IntersectionObserver) => {
+      if (entry) {
+        if (entry.isIntersecting) {
+          const imgElement = entry.target as HTMLImageElement;
+          imgElement.src = src;
+
+          imgElement.onload = () => {
```

Comment:
If `imgElement.complete` is true, we don't have to register `onload` handler. Because it means an image has already been loaded and cached in browser memory.

```typescript
if (imgElement.complete) {
  /* ... */
} else {
  imgElement.onload = () => {
    /* ... */
  };
}
```

Reply:
@HoseungJang 
I've tested it, and the moment an `imgElement` is exposed to the viewport using the `intersection observer` and the `src` attribute is assigned, whether it's cached in the browser or not, the `complete` attribute becomes false.

If you don't use "imgElement", but instead use "new Image()", you get the same result. 

So, I don't know about other cases, but for the `useLazyImage` hook that utilizes the `intersection observer`, I don't think the logic to check `complete` is necessary.

Reply:
In Safari I didn't encounter the problem. It works as I said. I think there is a bug in Chrome.

Reply:
@HoseungJang 
As you said, it doesn't work correctly only in chrome. 
I have confirmed that it works fine in safari and firefox.

Reply:
Oh, that's right. I've added that code. Thanks. 
>And I think there is no reason not to add this logic even though complete property doesn't work correctly (only) in Chrome.